### PR TITLE
Added a Blog section

### PR DESCRIPTION
**Pull request by HashirGJ8842**
Added a blogs section for the website. Currently, only some members of the club are allowed to add Blogs in the website.

### CODE REVIEW

File: pages/blogs/add.vue [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+            {
+              names = snapshot.val()[i].name;
+            }
+          }
+        });
+        this.name = names
+        this.uid = p
+      },
+			writeBlog() {
+        db.ref(`blogs/main/${this.count}`).set({
```
Comment (iamareebjamal) [inline pages/blogs/add.vue]: Everything is saved in blogs/main. Which will override every new blog

File: pages/blogs/add.vue [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+            title: this.title,
+            subtitle: this.subtitle,
+            content: this.content,
+            name: this.name,
+            uid: this.uid,
+            link: this.link
+        });
+        db.ref('blogs/blogsCount').set({
```
Comment (iamareebjamal) [inline pages/blogs/add.vue]: There is no need of this counter

File: pages/blogs/index.vue [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    components: {
+      PageLoader
+    },
+    data(){
+        return{
+            image: "", 
+            title: null,
+            name: null,
+            content: null,
+            id: null,
```
Comment (iamareebjamal) [inline pages/blogs/index.vue]: Above 5 properties should not be used

File: pages/blogs/index.vue [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+            /* eslint-disable */
+            const bloggerRef = await db.ref('blogs/bloggers');
+            let bloggers = [];
+            let p = 0;
+            bloggerRef.on('value', function(snapshot){
+              for(const i in snapshot.val())
+              {
+                bloggers.push(i)
+              }
+            })
```
Comment (iamareebjamal) [inline pages/blogs/index.vue]: Not needed. Use vuexfire

File: pages/blogs/index.vue [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+                  {
+                    blog.push({
+                        id: i,
+                        name: snapshot.val()[i].name,
+                        title: snapshot.val()[i].title,
+                        subtitle: snapshot.val()[i].subtitle,
+                        date: snapshot.val()[i].date,
+                        image: snapshot.val()[i].link,
+                        uid: snapshot.val()[i].uid}
+                        );
```
Comment (iamareebjamal) [inline pages/blogs/index.vue]: Not needed. Use vuexfire

File: pages/blogs/_id.vue [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+            title = snapshot.val().title;
+            content = snapshot.val().content;
+            subtitle = snapshot.val().subtitle;
+            image = snapshot.val().link;
+        });
+        this.name = name;
+        this.title = title;
+        this.content = content;
+        this.subtitle = subtitle;
+        this.image = image;
```
[the remark this answers is not in the document]
Comment (iamareebjamal) [reply pages/blogs/_id.vue]: This won't be needed when you use vuexfire

File: pages/blogs/add.vue [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+              <li>
+                list item 1
+              </li>
+              <li>
+                list item 2
+              </li>
+              <li>
+                list item 3
+              </li>
+            </ul>`,
```
[the remark this answers is not in the document]
Comment (iamareebjamal) [reply pages/blogs/add.vue]: And a single field will be used instead and not 4-5 fields like image, list, etc

**Comment by HashirGJ8842** [comment]
Thanks for the detailed review. I will be resolving those soon.

**Comment by iamareebjamal** [comment]
There are still linting issues

**Comment by HashirGJ8842** [comment]
Sorry, i didn't get that. Is there an issue with the structure or the indentation of the code?

**Comment by iamareebjamal** [comment]
Yes, formatting of code isn't proper

**Comment by HashirGJ8842** [comment]
Okay, thanks, will be resolving it now.

File: pages/blogs/add.vue [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+  methods: {
+    ...mapActions('blogs', ['setBlogsRef', 'setTeamRef']),
+    setUser() {
+      const p = this.user.uid
+      this.team.forEach(user => {
+        if (p === user.uid) {
+          this.currentUser.name = user.name
+          this.currentUser.uid = user.uid
+        }
+      })
```
Comment (iamareebjamal) [inline pages/blogs/add.vue:74]: This shouldn't be done here, this should be done in cloud functions, or anyone can set any user ID from JS

File: store/blogs.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+const bloggersRef = db.ref('blogs/bloggers')
+const blogsRef = db.ref('blogs/main')
+const userRef = db.ref('team/current/members')
+
+export const state = () => ({
+  bloggers: [],
+  blogs: [],
+  blog: [],
+  team: []
```
Comment (iamareebjamal) [inline store/blogs.js:12]: This is already present in teams store, and hence is redundant

File: store/contributions.js [hunk 10]
```diff
@@ -1,5 +1,5 @@
-import { firebaseAction } from 'vuexfire'
 import { db } from '@/plugins/firebase'
+import { firebaseAction } from 'vuexfire'
```
Comment (iamareebjamal) [inline store/contributions.js:2]: Revert the changes made in files not touched in this PR

File: pages/projects/index.vue [hunk 11]
```diff
@@ -103,7 +103,7 @@ export default {
     }
   },
   computed: {
-    projectLink() {
+    project() {
```
Comment (iamareebjamal) [inline pages/projects/index.vue:106]: Revert

File: pages/blogs/index.vue [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+  },
+  methods: {
+    ...mapActions('blogs', ['setBlogsRef']),
+    ...mapActions('blogs', ['setBloggersRef']),
+    deleteBlog(id) {
+      db.ref(`blogs/main/${id}`).remove()
+    },
+    addLink() {
+      return '/blogs/add'
+    },
```
Comment (iamareebjamal) [inline pages/blogs/index.vue:101]: This is static and hence no need to make it a method

File: pages/blogs/add.vue [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+      this.preview = true
+    },
+    writeBlog() {
+      db.ref(`blogs/main/${this.blogs.length}`).set({
```
Comment (iamareebjamal) [inline pages/blogs/add.vue:78]: Use push and see how it is done in news section. Using blog.length will overwrite any blog on race condition and will also create auth problems

File: store/contributions.js (hunk 10 above)
Comment (HashirGJ8842) [reply store/contributions.js:2]: Okay

File: pages/blogs/index.vue [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+    this.userUid()
+  },
+  updated() {
+    this.userUid()
+  },
+  methods: {
+    ...mapActions('blogs', ['setBlogsRef']),
+    ...mapActions('blogs', ['setBloggersRef']),
+    deleteBlog(id) {
+      db.ref(`blogs/main/${id}`).remove()
```
Comment (iamareebjamal) [inline pages/blogs/index.vue:97]: Should be in store as well

File: pages/blogs/index.vue [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+    },
+    addLink() {
+      return '/blogs/add'
+    },
+    blogLink(id) {
+      return `/blogs/${id}`
+    },
+    async userUid() {
```
Comment (iamareebjamal) [inline pages/blogs/index.vue:105]: Should be done using computed property, no need to await on things

File: pages/blogs/index.vue [hunk 16]
```diff
@@ -0,0 +1,142 @@
...
+          if (blogger['.key'] === User.uid) {
+            this.buttonShow = true
+            return true
+          }
+        })
+      } else {
+        this.buttonShow = false
+        return false
+      }
+    },
```
Comment (iamareebjamal) [inline pages/blogs/index.vue:119]: This should be broken in blog component and then use computed property to shoe the icon or not.

Secondly, there is no need to use a blogger collection. Just uid in blog is enough

File: pages/blogs/index.vue [hunk 17]
```diff
@@ -0,0 +1,142 @@
...
+    ifBlogOwner(id) {
+      if (this.user) {
+        if (this.user.uid === id) {
+          return true
+        }
+      }
+      return false
+    }
```
Comment (iamareebjamal) [inline pages/blogs/index.vue:127]: Same as above, make a blog component and use a computed property. Methods are used to do actions.

File: pages/blogs/add.vue [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+  data() {
+    return {
+      value: false,
+      title: '',
+      subtitle: '',
+      content: '',
+      count: 0,
+      preview: false,
+      link: '',
+      currentUser: {}
```
Comment (iamareebjamal) [inline pages/blogs/add.vue:53]: No need to store as data, this is a read only attribute. Data is used for reactive read-write attributes

File: pages/blogs/add.vue [hunk 19]
```diff
@@ -0,0 +1,107 @@
...
+export default {
+  data() {
+    return {
+      value: false,
+      title: '',
+      subtitle: '',
+      content: '',
+      count: 0,
+      preview: false,
+      link: '',
```
Comment (iamareebjamal) [inline pages/blogs/add.vue:52]:
```suggestion
      blog: {
	      title: '',
	      subtitle: '',
	      content: '',
	      count: 0,
	      preview: false,
	      link: ''
      },
```

Comment (iamareebjamal) [reply pages/blogs/add.vue:52]: See how it is done in news section

**Comment by HashirGJ8842** [comment]
@iamareebjamal, please review

**Comment by HashirGJ8842** [comment]
@iamareebjamal, @Haider8, please review it
